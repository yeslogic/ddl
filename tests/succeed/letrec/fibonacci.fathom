letrec
    fibonacci : U32 -> U32
    = fun (x : U32) => match x {
        0 => 0,
        1 => 1,
        _ => (fibonacci (x - 1)) + (fibonacci (x - 2)),
    };
fibonacci
