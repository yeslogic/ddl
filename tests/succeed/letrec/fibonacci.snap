stdout = '''
letrec 
    fibonacci : U32 -> U32 = fun x => match x {
        0 => 0,
        1 => 1,
        _ => fibonacci (x - (1 : U32)) + fibonacci (x - (2 : U32)),
    },
;
fibonacci : U32 -> U32
'''
stderr = ''
